Source/Parallelization/WarpXComm.cpp

```diff
@@ -548,14 +549,22 @@ WarpX::FillBoundaryE (int lev, PatchType patch_type, IntVect ng)
 {
     if (patch_type == PatchType::fine)
     {
-        if (do_pml && pml[lev]->ok())
-        {
-            pml[lev]->ExchangeE(patch_type,
-                                { Efield_fp[lev][0].get(),
-                                  Efield_fp[lev][1].get(),
-                                  Efield_fp[lev][2].get() },
-                                do_pml_in_domain);
-            pml[lev]->FillBoundaryE(patch_type);
+        if (do_pml) {
+            if (pml[lev] && pml[lev]->ok())
+            {
+                pml[lev]->ExchangeE(patch_type,
+                                    { Efield_fp[lev][0].get(),
+                                      Efield_fp[lev][1].get(),
+                                      Efield_fp[lev][2].get() },
+                                    do_pml_in_domain);
+                pml[lev]->FillBoundaryE(patch_type);
+            }
+#ifdef WARPX_DIM_RZ
+            if (pml_rz[lev])
+            {
+                pml_rz[lev]->FillBoundaryE(patch_type);
```

Comment:
Why do we not need to exchange data between the valid domain and the PML, as we do in Cartesian geometry (see call to `ExchangeE` function above)?

Reply:
This because of the way that the RZ pml's operate. The interactions between the main grid and the pml happens elsewhere. When the RZ pml fields are pushed, they depend on the fields from the main grid. Then the pml fields act back on the main grid in the routine `PML_RZ::ApplyDamping`.